### PR TITLE
Miband integration

### PR DESCRIPTION
**Pull request by bigdigital**
This branch allows direct communication between xdrip and miband so it is possible to display bg values and bg graph directly on the watch screen. It also allow to send and snooze alarms.  Also it is possible to control watch settings via xdrip. 
The current version supports miband 2,3 (only notifications) and miband 4.
More info available here 
https://bigdigital.home.blog/2020/01/23/future-of-the-xdrip-miband-project/

The code was tested for 3 months, and all major issues were fixed. If you have any questions regarding code, let me know.
Regards,
Artem

### CODE REVIEW

File: app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java [hunk 1]
```diff
@@ -417,14 +442,18 @@ protected void onResume()
             LocationHelper.requestLocationForBluetooth(this); // double check!
         }
         PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(LeFunEntry.prefListener);
+        PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(MiBandEntry.prefListener);
         PreferenceManager.getDefaultSharedPreferences(this).registerOnSharedPreferenceChangeListener(BlueJayEntry.prefListener);
+        LocalBroadcastManager.getInstance(this).registerReceiver(mibandStatusReceiver,
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java:447]: Does this need an associated `unregisterReceiver` in `onPause()` ?

Comment (bigdigital) [reply app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java:447]: Yes, you right, this is a good suggestion to deregister this receiver when activity paused.

Comment (jamorham) [reply app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java:447]: If you can update the PR with this change I can approve it for merging. thanks

Comment (bigdigital) [reply app/src/main/java/com/eveningoutpost/dexdrip/utils/Preferences.java:447]: Ok, done

**Pull request merged**